cfme/physical/physical_storage.py

```diff
@@ -0,0 +1,231 @@
+
+        Args:
+            refresh (bool): Whether to perform the page refresh, defaults to False
+        """
+        view = navigate_to(self, "Details")
+        if refresh:
+            view.browser.refresh()
+            view.flush_widget_cache()
+
+    def _execute_button(self, button, option, handle_alert=False):
+        view = navigate_to(self, "Details")
+        view.toolbar.custom_button(button).item_select(option, handle_alert=handle_alert)
+        return view
+
+    def _execute_action_button(self, button, option, handle_alert=True, **kwargs):
+        target = kwargs.get("target", None)
+        provider = kwargs.get("provider", None)
+        desired_state = kwargs.get("desired_state", None)
+
+        view = self._execute_button(button, option, handle_alert=handle_alert)
+
+        if desired_state:
+            self._wait_for_state_change(desired_state, target, provider, view)
+        elif handle_alert:
+            wait_for(
+                lambda: view.flash.is_displayed,
+                message="Wait for the handle alert to appear...",
+                num_sec=5,
+                delay=2
+            )
+
+    def product_name(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Product Name")
+
+    def serial_number(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Serial Number")
+
+    def health_state(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Health State")
+
+    def enclosure_count(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Enclosure Count")
+
+    def drive_bays(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Drive Bays")
+
+    def uuid(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("UUID")
+
+    def description(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Description")
+
+    def _wait_for_state_change(self, desired_state, target, provider, view, timeout=300, delay=10):
+        """Wait for the Physical Storage to reach the desired state. This function waits an amount
+           of time due to wait_for.
+
+        Args:
+            desired_state (str): 'on' or 'off'
+            target (str): The name of the method that most be used to compare with the desired_state
+            provider (object): 'LenovoProvider'
+            view (object): The view that most be refreshed to verify if the value was changed
+            timeout (int): Specify amount of time (in seconds) to wait until TimedOutError is raised
+            delay (int): Specify amount of time (in seconds) to repeat each time.
+        """
+
+        def _is_state_changed():
+            self.refresh(provider, handle_alert=True)
```

Comment:
Question;
Where is the refresh method in `BaseEntity`?
you can use `provider.refresh_provider_relationships`
Also,

Required:
You can accept provider as mandatory so you can use `self.provider` here.